lib/std/once.nim

```diff
@@ -0,0 +1,31 @@
+import atomics
+import locks
+
+
+type
+  Once* = object
+    finished: Atomic[bool]
+    lock: Lock
+
+proc initOnce*(once: var Once) =
+  once.finished.store(false)
+  initLock(once.lock)
+
+template once*(alreadyExecuted: Once, body: untyped) =
```

Comment:
can a 0 arg version work eg:

```nim
var count = 0
for i in 0..<3:
  once: inc count
doAssert count == 1

Reply:
No, it doesn't work because of `lock`. You need to maintain a var lock for each instance. It hard to use `{.global.}` to achieve this(maybe pure Nim version lock could improve this).